### PR TITLE
publish contract to object instead of account

### PR DESCRIPTION
**Pull request by 0xaptosj**
some context https://github.com/aptos-labs/aptos-developer-discussions/discussions/279

New publishing UX becomes
1. when user first get started with the template, run `npm run move:publish` to publish contract to an object.
2. if user makes compatible change to the contract, run `npm run move:upgrade` to upgrade and keep the same contract address
3. if user makes non-compatible change to the contract, run `npm run move:publish` to publish contract to a new object.

### CODE REVIEW

**Review by kaw2k — APPROVED** [review]
Looks fine to me, but I have less context. What do you think @0xmaayan

File: templates/digital-asset-template/scripts/move/upgrade.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  if (!process.env.VITE_MODULE_ADDRESS) {
+    throw new Error(
+      "VITE_MODULE_ADDRESS variable is not set, make sure you have published the module before upgrading it",
+    );
+  }
+
+  const move = new cli.Move();
+
+  move
+    .upgradeObjectPackage({
```
Comment (0xmaayan) [inline templates/digital-asset-template/scripts/move/upgrade.js:18]: this is duplicated with the publish script. can we somehow merge it? maybe for the publish command, call the upgrade command an then run a script to update the .env file?

File: templates/digital-asset-template/scripts/move/upgrade.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        // This is the address you want to use to create collection with, e.g. an address in Petra so you can create collection in UI using Petra
+        initial_creator_addr: process.env.VITE_COLLECTION_CREATOR_ADDRESS,
+        // Our contract depends on the token-minter contract to provide some common functionalities like managing refs and mint stages
+        // You can read the source code of it here: https://github.com/aptos-labs/token-minter/
+        // Please find it on the network you are using, This is testnet deployment
+        minter: "0x3c41ff6b5845e0094e19888cba63773591be9de59cafa9e582386f6af15dd490",
+      },
+      profile: process.env.VITE_APP_NETWORK,
+    })
+    .then(console.log);
```
Comment (0xmaayan) [inline templates/digital-asset-template/scripts/move/upgrade.js:33]: why we need to console log here?

Comment (0xaptosj) [reply templates/digital-asset-template/scripts/move/upgrade.js:33]: just to display the result

File: templates/digital-asset-template/scripts/move/upgrade.js (hunk 1 above)
Comment (0xaptosj) [reply templates/digital-asset-template/scripts/move/upgrade.js:18]: which part is duplicate? you mean the function parameter of `createObjectAndPublishPackage` and `upgradeObjectPackage` is the same?

Comment (0xmaayan) [reply templates/digital-asset-template/scripts/move/upgrade.js:18]: I mean the use of `upgradeObjectPackage` in both publish and upgrade scripts

File: templates/digital-asset-template/scripts/move/upgrade.js (hunk 2 above)
Comment (0xmaayan) [reply templates/digital-asset-template/scripts/move/upgrade.js:33]: dont think we need it? it outputs any cli response to the console already

File: templates/digital-asset-template/scripts/move/upgrade.js (hunk 1 above)
Comment (0xaptosj) [reply templates/digital-asset-template/scripts/move/upgrade.js]: `upgradeObjectPackage` is only used in upgrade script, not in publish script

File: templates/digital-asset-template/scripts/move/upgrade.js (hunk 2 above)
Comment (0xaptosj) [reply templates/digital-asset-template/scripts/move/upgrade.js]: remvoed

File: templates/digital-asset-template/scripts/move/publish.js [hunk 3]
```diff
@@ -4,32 +4,55 @@ const yaml = require("js-yaml");
 const cli = require("@aptos-labs/ts-sdk/dist/common/cli/index.js");
 
 const config = yaml.load(fs.readFileSync("./.aptos/config.yaml", "utf8"));
-const accountAddress =
-  config["profiles"][process.env.VITE_APP_NETWORK]["account"];
+const accountAddress = config["profiles"][process.env.VITE_APP_NETWORK]["account"];
```
Comment (0xmaayan) [inline templates/digital-asset-template/scripts/move/publish.js:7]: this needs to be changed, look at https://github.com/aptos-labs/create-aptos-dapp/pull/122

**Pull request merged**